src/cuda.jl

```diff
@@ -0,0 +1,15 @@
+
+@inline function Cassette.overdub(
+    ctx::RecordingCtx, ::Type{A}, ::UndefInitializer, dims...
+) where {A<:CUDA.CuArray}
+    ret = A(undef, dims...)
+    record_alloc!(ctx, ret)
+    return ret
+end
+
+@inline function Cassette.overdub(
+    ctx::ReplayCtx, ::Type{A}, ::UndefInitializer, dims...
+)::A where {A<:CUDA.CuArray}
+    scheduled = next_scheduled_alloc!(ctx) :: A
+    return scheduled
+end
```

Comment:
Should we make a macro that declares these two functions,
and then we can call it on

```
@make_autopreallocatable CUDA.Array
@make_autopreallocatable CUDA.CuArray
```


so as to remove the duplication of code between this and `Array`.
And so it is easy to add other types to it.

Reply:
With just two cases (and off the top of my head I can't think of another that exists in Julia that we'd want), I'm not sure its worth it. Could certainly add in the future.

Reply:
I think you mentioned applying it to wrapper array types, because that let you avoid Cassette related inference woes of having to decend through them?